package/gluon-mesh-vpn-tunneldigger/files/usr/bin/tunneldigger-watchdog

```diff
@@ -0,0 +1,44 @@
+#!/usr/bin/lua
+
+function restart_tunneldigger()
+	io.popen('logger -t tunneldigger-watchdog "Restarting Tunneldigger."')
+	os.execute("/etc/init.d/tunneldigger restart")
+end
+
+function read_pid_file()
+	local pid_file = io.open("/var/run/tunneldigger.mesh-vpn.pid", "r")
+	if not pid_file then
+		return nil
+	end
+	pid = pid_file:read("*l") 
+	pid_file:close()
+	return pid
+end
+
+function check_for_mesh_vpn_neighbours()
+	local handle = io.popen("batctl o", 'r')
+	if not handle then
+		for line in handle:lines() do
```

Comment:
This loop (and all nested code) is only reached in case your `handle` is `nil`. This is most likely not what you wanted to achieve.

Reply:
Ah thanks. Must have happen during the refractoring :)